Blackjack/InputParser.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+
+namespace Blackjack
+{
+    public static class InputParser
+    {
+        public static void ParseChoice(string v, object input)
+        {
+            throw new NotImplementedException();
+        }
+
+        public static int ParseChoice(string input)
+        {
+            var number = IsValid(input);
+            return number;
+        }
+
+        private static int IsValid(string input)
+        {
+            var isValidNumber = Int32.TryParse(input, out var number);
+            // if (number < 0 || number > 1) return ; // what to return if input not valid

```

Comment:
Refer to what you did for the input validation in Conway's

Reply:
Do you care if it's less than 0 or greater than 1? If you were to evaluate this predicate, you'd get `true` when the number isn't valid.

Or rather do you care that it's equal to 0 or equal to 1? Generally a predicate will read better if you evaluate the "positive" as `true` and the "negative" as false i.e. `IsValid` should evaluate to `true` if it is valid and `false` if it isn't. This is also nice as you can do simple returns like
```suggestion
            return (number == 0 || number == 1);
```

Presumably line 21 is commented out because you're not able to guarantee `number` exists. You'd need to check if `number` exists. Can you use string comparison here to save a `Parse` and save a check of if `number` exists?